CONTRIBUTING.md

```diff
@@ -137,9 +144,143 @@ The [packages/](packages/) directory contains the following public/published pac
 ### Internal packages
 
 The [packages/](packages/) directory contains the following internal packages:
+
 * [caliper-publish](packages/caliper-publish/): Utility CLI for publishing Caliper to NPM and DockerHub.
 * [caliper-tests-integration](packages/caliper-tests-integration/): Collection of CI integration tests.
 
+### Testing Methodologies
+
+This section provides important information about testing this project. It includes dependencies used for testing and detailed instructions on how to add conforming tests.
+
+#### Testing Dependencies and Their Use
+
+* [mockery](https://www.npmjs.com/package/mockery): Mockery is a simple module for mocking Node.js modules during testing. It allows you to replace real modules with mocks or stubs.
+  
+* [mocha](https://mochajs.org/): Mocha is a feature-rich JavaScript test framework that runs on Node.js and in the browser. It facilitates asynchronous testing, making it easy to write simple and flexible tests.
+
+* [chai](https://www.npmjs.com/package/chai): Chai is a BDD/TDD assertion library for Node.js and the browser. It can be paired with any JavaScript testing framework. We use it to create readable and expressive assertions.
+
+* [sinon](https://sinonjs.org/releases/v18/): Sinon is a standalone test spies, stubs, and mocks for JavaScript. It works with any test framework and integrates well with Mocha and Chai. We utilize Sinon for checking how functions are called during testing.
+
+* [sinon-chai](https://www.npmjs.com/package/sinon-chai): This library provides a set of custom assertions for using Sinon with Chai. It allows you to write more readable assertions for Sinon spies, stubs, and mocks.
+
+* [nyc](https://www.npmjs.com/package/nyc): NYC is a command-line utility for generating code coverage reports. It is often used with Mocha to ensure that tests cover as much code as possible.
+
+#### Example Test Case and Points to Note for Adding a Conforming Test
```

Comment:
I'm not sure I am keen on this section as it's presented here as I don't feel it's conveying the information that I think is required. I appreciate we may want to try to keep unit tests using the same style but unit testing can get complex with the way things get mocked out and so there will be all manner of different patterns you might employ to achieve this.

I'd like that we start what I would consider mandatory for every unit test
1. License Header
2. 'use strict'
3. Guidance on how to split unit tests into describe blocks and it statements and even describes within describes
4. The format of the text within a Describe and It statements as they need to flow as seen in the fabric tests. These all follow a pattern which should be iterated here.
5. Warn of the dangers of mocks that persist and affect other tests in unexpected ways, always run all unit tests before submitting a PR and check for left over `.only` code that will stop all tests from running
6. Refer to the Fabric Unit tests for examples of patterns of implementation

I'm not sure the rest provides any real value here as it will be dependent on what is required to make the unit test work for example how you use chai I don't think should be mandated as people may prefer to use `expect` rather than `should` and I prefer to use expect a lot of the time as well now

Reply:
I agree this serves the purpose of the section better